baseapp/baseapp.go

```diff
@@ -718,35 +720,85 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 	// Result if any single message fails or does not have a registered Handler.
 	result, err = app.runMsgs(runMsgCtx, msgs, mode)
 
-	if err == nil {
+	// Case 1: the msg errors and the post handler is not set.
+	if err != nil && app.postHandler == nil {
+		return gInfo, nil, events, priority, err
```

Comment:
Before this was returning `anteEvents`

Reply:
correct, it should be only `anteEvents` here given that the tx failed